libretro/libretro.cpp

```diff
@@ -678,6 +920,22 @@ void retro_init(void)
    else
       log_cb = NULL;
 
+   const char *dir = NULL;
+
+   if (environ_cb(RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY, &dir) && dir)
+      snprintf(retro_system_directory, sizeof(retro_system_directory), "%s", dir);
+   else
+      snprintf(retro_system_directory, sizeof(retro_system_directory), "%s", ".");
+
+   if (environ_cb(RETRO_ENVIRONMENT_GET_SAVE_DIRECTORY, &dir) && dir)
+      snprintf(retro_save_directory, sizeof(retro_save_directory), "%s", dir);
+   else
+      snprintf(retro_save_directory, sizeof(retro_save_directory), "%s", ".");
+
+   // State that SNES9X supports achievements.
+   bool achievements = true;
+   environ_cb(RETRO_ENVIRONMENT_SET_SUPPORT_ACHIEVEMENTS, &achievements);
```

Comment:
Since we don't have the memory mapping feedback I think we also need to set this to false, right?

Reply:
not sure what you mean with memory mapping feedback, achievements are read only (we only read memory)